### DISCUSSION TITLE
ROS2 over wifi?

### DISCUSSION THREAD

**Discussion by mkopack73** [iRobotEducation/create3_docs#57] [Questions] [answered]
Ok, I'm confused on how to get my Linux box with ROS2 Galactic to communicate and control the Create3 over wifi.

I've set both up to use ROS_DOMAIN_ID=17

I tried directly connecting over wifi to the  roomba, but running the  create3_dance  program from the examples just reports: 

[INFO] [1652575139.852565506] [dance_command_publisher]: service not available, waiting again...

I then went into the roomba's configuration and had it connect to my home Wifi, and connected my computer to the same network. Again trying to run the create3_dance program gave the same error.

I tried pinging the hostname I set the roomba up as: "iRobot-9FFF.local"  but got:

 ping: iRobot-9FFF.local: Name or service not known

Trying to do a: "ros2 topic list" just gives me back:
/parameter_events
/rosout

So it doesn't seem like it's connected and getting any topics from the roomba itself.

What am I doing wrong? Or is what I'm trying to do not possible over Wifi?  It'll be a few days before I can get a USB-C->USB-C cable to hook up one of my Pi 4's directly...

**Comment by shamlian** [comment, marked as answer]
Hi there,
It seems to me like what you are doing should work, but given that you can't ping the robot, it looks like it's not getting on your network.

Can you put your robot in [Bluetooth mode](https://iroboteducation.github.io/create3_docs/hw/adapter/#usbble-toggle) (the blue LED on the adapter board, D6, should be on) and then run the "create3_ip_address.py" example at [python.irobot.com](https://python.irobot.com/) to see if your robot got an IP address? Alternatively, if you have access to your Wi-Fi router, you should be able to see the robot's IP address in its config page. (You'll have to take the robot out of Bluetooth mode once you want to connect over USB.)

If you don't see that the robot is getting an IP address, try [provisioning](https://edu.irobot.com/create-3-kick-off) again. Note that the Create 3 robot's radio can only connect to 2.4 GHz networks. If the robot is getting an IP address, and you can't ping it, I'm not sure what's wrong, but I would be suspicious of your network environment. (Does it support ICMP and multicast?)

If your robot _is_ getting an IP address and you can ping it from your computer, you should be able to get topics with a "ros2 topic list" from that computer. If that doesn't work, my first guess would be that your RMW selection on your computer and robot don't match. My next guess would be that your network doesn't support multicast, and you'll have to follow steps on [this page](https://iroboteducation.github.io/create3_docs/setup/xml-config/).

Best of luck; let us know how it goes.

**Comment by mkopack73** [comment]
I figured it out...  It apparently REALLY doesn't like me setting the DOMAIN_ID to something other than 0. When I set it to 17 I couldn't even get it to do a ROS2 Topic list - it would just hang. 

Set everything back to using 0, and suddenly it started working.

Thanks for the link to the Ubuntu page. That did help some. I def had to set the config.xml file up for the DDS because the VM I'm running in exposes a different interface name due to the bridged adaptor.

**Reply by shamlian** [reply]
The Create 3 robot is tested and absolutely works with different domain IDs, but IMHO (this is not iRobot's perspective but my own), ROS 2 network setup in general definitely leaves something to be desired, from a UX perspective. I would recommend getting things working to your satisfaction and only messing with more complicated network setup if you have to.

**Comment by mkopack73** [comment]
Oh, and the reason I wasn't finding it via ping was I was trying to use:
ping: iRobot-9FFF.local

it didn't like that... Had to use
ping: iRobot-9FFF 
or 
ping: iRobot-9FFF.home
and then it worked and showed the IP address resolution and got me back responses.

**Reply by shamlian** [reply]
Interesting; what type of machine are you using?
